packages/core/src/core/executer.ts

```diff
@@ -49,17 +50,45 @@ export class Executer {
         process.on('beforeExit', this.closeChild);
         process.on('exit', this.closeChild);
 
-        child.on('exit', (code) => {
+        child.once('exit', (code) => {
             this.logger.info(`"${name}" PID ${child.pid} ${!code ? 'done' : `exit ${code}`}`);
             this.child = undefined;
         });
     }
 
     async closeChild() {
-        while (this.child && !this.child.killed) {
-            this.child.kill();
-            await delay(50);
+        if (!this.child) {
+            return;
+        }
+
+        const child = this.child;
+
+        const exitPromise = new Promise<void>((resolve) => {
+            child.once('exit', () => {
+                resolve();
+            });
+        });
+
+        try {
+            await this.terminateChild(child);
+            await exitPromise;
+        } finally {
+            this.child = undefined;
+        }
+    }
+
+    private async terminateChild(child: ExecaChildProcess) {
+        const maxRetries = 6;
+        let retries = 0;
+
+        while (child && !child.killed && retries < maxRetries) {
+            child.kill();
+            await delay(30);
+            retries++;
+        }
+
+        if (child && !child.killed) {
+            this.logger.warn(`Child process PID ${child.pid} failed to close after ${maxRetries} attempts`);
```

Comment:
If there is no guarantee that the process has been closed, it is meaningless to restart the next application for applications that include port monitoring or other resource occupation.

So I suggest that a warning should be given in the loop, and the loop will not terminate before the process is closed.